src/core/border_router/routing_manager.cpp

```diff
@@ -1198,6 +1189,27 @@ bool RoutingManager::NetworkDataContainsOmrPrefix(const Ip6::Prefix &aPrefix) co
     return contain;
 }
 
+// Update the `mRouterAdvMessage` with given Router Advertisement message.
+// Returns a boolean which indicates whether there are changes of `mRouterAdvMessage`.
+bool RoutingManager::UpdateRouterAdvMessage(const RouterAdv::RouterAdvMessage &aRouterAdvMessage)
+{
+    RouterAdv::RouterAdvMessage oldRouterAdvMessage;
```

Comment:
```suggestion
    bool changed = (aRouterAdvMessage != mRouterAdvMessage);
    VerifyOrExit(changed);
```

Save the copying `oldRouterAdvMessage = mRouterAdvMessage` ?

Reply:
Please notice that we don't always set `mRouterAdvMessage` to `aRouterAdvMessage`: we set `mRouterAdveMessage` to default values when `aRouterAdvMessage` has zero Router Lifetime.

Reply:
SGTM